pco/src/mode.rs

```diff
@@ -46,15 +46,32 @@ pub enum Mode<L: Latent> {
   /// Formula: (bin.lower + offset) * mode.base +
   /// (adj_bin.lower + adj_bin.offset) * machine_epsilon
   FloatMult(L),
+  /// This mode decomposes a float `x` with precision `p` into the pair `(y, m)`, where:
+  /// * `y` is the float with precision `p - k` obtained by truncating the least-significant `k`
+  ///   bits from the significand of `x`
+  /// * `m` is a `k`-bit unsigned integer consisting of the bits removed to form `y`.
+  ///
+  /// Here `k` is a parameter of the mode, and `p` is deduced (by table lookup) from
+  /// the underlying floating point type.
+  ///
+  /// Each number is compressed as
+  /// * which bin the latent-bits representation of `y` is in,
+  /// * the offset of the latent-bits representation of `y` in that bin,
+  /// * which bin `m` is in, and
+  /// * the offset of `m` in that bin.
+  ///
+  /// Formula: F::from_latent_bits(((y_bin.lower + y_bin.offset) << mode.k) + m)
+  FloatDecomp(L),
 }
 
 impl<L: Latent> Mode<L> {
   pub(crate) fn n_latent_vars(&self) -> usize {
     use Mode::*;
 
     match self {
-      Classic => 1,
-      FloatMult(_) | IntMult(_) => 2,
+      Classic => 1,  // delta_order
+      FloatMult(_) | IntMult(_) => 2,  // delta_order, mode.base
```

Comment:
You can lump these cases together with another `|`

Reply:
Does that feel more readable to you?  I separate them out so that I could add a comment on each line saying what the parameters are.  I'll defer to you if you prefer the one-line, no-comment version though.

Reply:
I didn't realize that's what you're doing - if you want to label each one, that's fine, but they'd be something like e.g. "multiplier, ULPs adjustment" or "mult, adj" for the *Mult modes.

Like for int mult mode, the compression algorithm effectively says that there are some distributions `L_0`, `L_1`  (the latent (random) variables) such that each number is `x = l_0 * base + l_1`, `l_0 ~ L_0`, `l_1 ~ L_1`. The instances `l_0, l_1` are the latents.

Reply:
> I didn't realize that's what you're doing - if you want to label each one, that's fine, but they'd be something like e.g. "multiplier, ULPs adjustment" or "mult, adj" for the *Mult modes.
> 
> Like for int mult mode, the compression algorithm effectively says that there are some distributions `L_0`, `L_1` (the latent (random) variables) such that each number is `x = l_0 * base + l_1`, `l_0 ~ L_0`, `l_1 ~ L_1`. The instances `l_0, l_1` are latents drawn from the latent (random) variables.


Ah, I didn't realize that's what `latent_vars` meant in this case, thanks!

I do think "latents" currently means several different things, and even if I'm wrong about that, it might lead to some improvements to the docs :) so I'll say a bit:

In my possibly-mistaken understanding of the codebase, "latent" is used to mean:
* A thing of type `L`, regardless of its meaning (e.g.: `to_latent_bits()`)
* The uint that corresponds to an input datum under the order-preserving injection `to_latent_ordered()`
* An element of the tuple returned by `split_latents` (an injective map that carries a single datum (e.g. an `F`) to possibly-multiple `L`s)

From talking with you, I understand there is also another notion of "latent," though I haven't seen it in the codebase:
* If we treat compression as a source coding problem, where the data are modelled as i.i.d., then pco is optimal or near-optimal in the case where the true data distribution is a certain latent variable model where the things called `l0` and `l1` in the readme are the latent variables.

I currently think of that last one as being your personal framing of pco, and it's a reasonable one.  But it seems not "official" in the sense that I don't see it reflected in the code or laid out directly in the docs, and I wouldn't expect most users to prefer that framing or arrive at it on their own.  (Maybe I'm wrong, I don't know much about the culture of compression libraries.) Also, that latent variable model is unusual in that the latents are a deterministic function of the observed value.  That's part of why I don't naturally think of pco in terms of a latent variable model, I insetad think of it in terms of carefully designed data transformations that separate out compressible components.

WDYT?

Reply:
Hm yeah, I probably need to understand the delta stuff and the tANS stuff more before I can make more informed comments.  One thing that crossed my mind and is maybe independent of that, wdyt having the crate-global name of the type be something like `UInt`, and then each component of the code (e.g., each `Mode`) can have a local typedef?  E.g., the existing modes could have a member `type Latent = UInt`?  This is maybe not so much a suggestion for The One Name to use, but a suggestion that maybe allowing local aliases per mode could help clarify things.

Reply:
Oh yeah this was never for consideration in this PR.  I was suggesting in case it gave you ideas; if it doesn't sound right to you then I definitely defer to that because I don't have enough codebase familiarity to be confident that a refactor suggestion is good.